pkg/agent/agentclient/client.go

```diff
@@ -27,11 +27,54 @@ import (
 	"sigs.k8s.io/apiserver-network-proxy/konnectivity-client/proto/client"
 )
 
+// connContext tracks a connection from agent to node network.
+type connContext struct {
+	conn      net.Conn
+	cleanFunc func()
+	dataCh    chan []byte
+	cleanOnce sync.Once
+}
+
+func (c *connContext) cleanup() {
+	c.cleanOnce.Do(c.cleanFunc)
+}
+
+type connectionManager struct {
+	mu          sync.RWMutex
+	connections map[int64]*connContext
+}
+
+func (cm *connectionManager) Add(connID int64, ctx *connContext) {
+	cm.mu.Lock()
+	defer cm.mu.Unlock()
+	cm.connections[connID] = ctx
+}
+
+func (cm *connectionManager) Get(connID int64) (*connContext, bool) {
+	cm.mu.RLock()
+	defer cm.mu.RUnlock()
+	ctx, ok := cm.connections[connID]
```

Comment:
nit: Could do just `return cm.connections[connID]`. May affect readability so I'm not too sure